refactor(linter): Always pass the Rule itself to the DefaultRuleConfig.

connorshea opened the pull request:
Rather than passing the inner config.

With this change, we're consistent across all rules that use DefaultRuleConfig now. We can also replace the rule name with Self in the future.

connorshea commented:
I'm pretty sure most of Copilot's comments here were wrong, but please check.

camc314 commented:
### Merge activity

* **Dec 19, 1:53 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 19, 1:53 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
refactor(linter): Always pass the Rule itself to the DefaultRuleConfig. (#17106)

Rather than passing the inner config.

With this change, we're consistent across all rules that use DefaultRuleConfig now. We can also replace the rule name with Self in the future.
